daemon/emer-persistent-cache.c

```diff
@@ -1086,6 +1104,28 @@ emer_persistent_cache_read (EmerPersistentCache *self,
       local_variants[i] = regularize_post_storage (curr_variant);
     }
 
+  if (corrupt_data)
+    {
+      /* Somehow the circular file contains corrupt data, meaning that we
+       * can't trust its contents anymore so we clean up and return an error. */
+      gsize j = 0;
+
+      while (j < num_elems)
+        g_bytes_unref (elems[j++]);
```

Comment:
The bytes in `elems` with index between `0` and `i - 1` have already been unreferenced in the for loop above before encountering the corrupt data. So I think that `j` should be initialized to the value of `i` before the cycle, not `0`.